editor/store/selectors.js

```diff
@@ -1111,15 +1111,132 @@ export function getNotices( state ) {
 }
 
 /**
- * Resolves the list of recently used block names into a list of block type settings.
- *
- * @param {Object} state Global application state
- *
- * @returns {Array} List of recently used blocks.
+ * An item that appears in the inserter. Inserting this item will create a new
+ * block. Inserter items encapsulate both regular blocks and reusable blocks.
+ * 
+ * @typedef {Object} Editor.InserterItem
+ * @property {string}   id                Unique identifier for the item.
+ * @property {string}   name              The type of block to create.
+ * @property {Object}   initialAttributes Attributes to pass to the newly created block.
+ * @property {string}   title             Title of the item, as it appears in the inserter.
+ * @property {string}   icon              Dashicon for the item, as it appears in the inserter.
+ * @property {string}   category          Block category that the item is associated with.
+ * @property {string[]} keywords          Keywords that can be searched to find this item.
+ * @property {boolean}  isDisabled        Whether or not the user should be prevented from inserting this item.
+ */
+
+/**
+ * Given a regular block type, constructs an item that appears in the inserter.
+ * 
+ * @param {Object}                state             Global application state.
+ * @param {string[]|boolean}      enabledBlockTypes Enabled block types, or true/false to enable/disable all types.
+ * @param {Object}                blockType         Block type, likely from getBlockType().
+ * @returns {Editor.InserterItem}                   Item that appears in inserter.
+ */
+function buildInserterItemFromBlockType( state, enabledBlockTypes, blockType ) {
+	if ( ! enabledBlockTypes || ! blockType ) {
+		return null;
+	}
+
+	const blockTypeIsDisabled = Array.isArray( enabledBlockTypes ) && ! enabledBlockTypes.includes( blockType.name );
```

Comment:
ES6 prototype methods are not polyfilled (e.g. `Array.from` is fine, but `Array#includes` is not).

This will error in IE11.

Reply:
👌 good to know. Looks like https://github.com/WordPress/gutenberg/pull/5528 addresses this.

Reply:
Could we polyfill prototype methods?

Reply:
Context at #746

tl;dr: Was waiting for Babel 7, which I didn't expect to take so long to release 😄